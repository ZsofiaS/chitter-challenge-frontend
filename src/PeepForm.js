import React, { Component } from 'react';

class PeepForm extends Component {
  constructor() {
    super();
    this.state = {
      userName: '',
      peep: '',
      newPeep: {}
    }
  }
  nameChangeHandler = (e) => {
    this.setState({
      userName: e.target.value
    })
  }
  peepChangeHandler = (e) => {
    this.setState({
      peep: e.target.value
    })
  }
  submitHandler = (e) => {
    e.preventDefault();
    let APIurl = 'https://chitter-backend-api-v2.herokuapp.com/peeps';

    fetch(APIurl,
      {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': 'Token token=a_valid_session_key',
        },
        body:
          `{"peep": {
              "user_id":1,
              "body":${this.state.peep}
              }
            }`
      })
      .then((response) => response.json())
      .then((data) => console.log(data))
    }

  render() {
    return (
      <form onSubmit={console.log("submitted")}>
        <input type="text" placeholder="Your name here..." onChange={this.nameChangeHandler} />
        <input type="text" placeholder="Your peep here..." onChange={this.peepChangeHandler} />
        <input type="submit" />
      </form>
    )
  }
}

export default PeepForm;
