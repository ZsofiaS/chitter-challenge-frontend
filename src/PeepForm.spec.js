import React from 'react';
import renderer from 'react-test-renderer';

import PeepForm from './App';

describe('PeepForm', () => {
  it('snapshot renders', () => {
    const tree = renderer.create(<PeepForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
