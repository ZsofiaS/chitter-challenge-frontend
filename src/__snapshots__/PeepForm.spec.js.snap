// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeepForm snapshot renders 1`] = `
<div
  className="App"
>
  <h1>
    Loading...
  </h1>
  
  <ul
    className="peepList"
  />
  <input
    id="nameInput"
    onChange={[Function]}
    placeholder="Your name..."
    type="text"
    value=""
  />
  <input
    id="passwordInput"
    onChange={[Function]}
    placeholder="Your password..."
    type="password"
    value=""
  />
  <button
    className="button shortButton"
    id="loginSubmit"
    onClick={[Function]}
  >
    Login
  </button>
  <button
    className="button shortButton"
    id="signupSubmit"
    onClick={[Function]}
  >
    Signup
  </button>
</div>
`;
